spec/routing/uploads_spec.rb

```diff
@@ -0,0 +1,79 @@
+require "spec_helper"
+
+describe "the uploads" do
+  include Capybara::DSL
+
+  context "user not logged in" do
+    it "visits uploads_path" do
+      visit uploads_path
+
+      expect(current_path).to eq(root_path)
+    end
+
+    it "visits new_upload_path" do
+      visit new_upload_path
+
+      expect(current_path).to eq(root_path)
+    end
+
+    it "visits edit_upload_path" do
+      upload = create(:upload)
+
+      visit edit_upload_path(id: upload.id)
+
+      expect(current_path).to eq(root_path)
+    end
+  end
+
+  context "user logged in" do
+    before { user = create(:user)
```

Comment:
Avoid using {...} for multi-line blocks.<br>Block body expression is on the same line as the block start.